packages/template-retail-react-app/app/components/store-locator/store-locator-content.jsx

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useState, useEffect} from 'react'
+import {useIntl} from 'react-intl'
+import PropTypes from 'prop-types'
+import {
+    Heading,
+    Button,
+    InputGroup,
+    Accordion,
+    AccordionItem,
+    Select,
+    Box,
+    Input
+} from '@salesforce/retail-react-app/app/components/shared/ui'
+import StoresList from '@salesforce/retail-react-app/app/components/store-locator/stores-list'
+import {Controller, useForm} from 'react-hook-form'
+import {
+    SUPPORTED_STORE_LOCATOR_COUNTRIES,
+    DEFAULT_STORE_LOCATORY_COUNTRY,
+    DEFAULT_STORE_LOCATORY_POSTAL_CODE
+} from '@salesforce/retail-react-app/app/constants'
+
+const StoreLocatorContent = () => {
+    const [searchStoresParams, setSearchStoresParams] = useState({})
+    const intl = useIntl()
+    var formattedStoreLocatorCountries = SUPPORTED_STORE_LOCATOR_COUNTRIES.map(
+        ({countryCode, countryName}) => {
+            return {countryCode: countryCode, countryName: intl.formatMessage(countryName)}
+        }
+    )
+    var defaultCountryCode = formattedStoreLocatorCountries.find(
+        (obj) => obj.countryName == intl.formatMessage(DEFAULT_STORE_LOCATORY_COUNTRY)
+    ).countryCode
+    const formProps = useForm({
+        mode: 'onChange',
+        reValidateMode: 'onChange',
+        defaultValues: {
+            countryCode: defaultCountryCode
+        }
+    })
+    const {control, handleSubmit} = formProps
+    useEffect(() => {
+        setSearchStoresParams({
+            countryCode: defaultCountryCode,
+            postalCode: DEFAULT_STORE_LOCATORY_POSTAL_CODE
+        })
+    }, [])
```

Comment:
Typically `useEffect` is for something asynchronous. In this case, we can find the stores params immediately. So I'd replace it and call `useState` with initial value like this:

```suggestion
    const [searchStoresParams, setSearchStoresParams] = useState({
        countryCode: defaultCountryCode,
        postalCode: DEFAULT_STORE_LOCATORY_POSTAL_CODE
    })
```

Reply:
Thank you! [The useState is now initialized with the default values directly](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1827/files#diff-2e3360613f21b8de36d12d3ff69d7d5fd7b7d92fdd6503abe7eef6234b675fc1R56).